### ISSUE TITLE
Prepare release 3.12.0

### ISSUE THREAD

**Issue by pljones** [jamulussoftware/jamulus#3443]
**Target timeline**
| Phase | Date |
|--------|--------|
| Scheduled feature freeze / Start of translation process | 06.07.2025 |
| Targeted translation completion date: | Early 2025 |
| Approximate release date: | Early 2026 |
| Current state: | Released | 

#### Release aims

We're currently looking forwards to Jamulus 4, which will make some larger scale changes to parts of Jamulus than previous releases have.  So we're aiming to fix as many known issues as we can in this coming release, so there's a stable version of Jamulus available whilst work progresses on the changes.

**Checklist**
- [ ] Assign this issue to the release shepherd who is in charge of managing this checklist.
- [x] Make sure `ChangeLog` has the agreed next release version number.
- [x] Pin this issue
- [x] Ensure that all issues/PR targeted for this release are done by checking the Project board with [the appropriate filter for this release](https://github.com/orgs/jamulussoftware/projects/5/views/5). Remind main developers to review entries in *Waiting on team* state.
- [x] Agree to de-tag unfinished Issues/PRs.
- [x] Declare a freeze for code and website by updating this Issue and adding a comment. PRs can still be worked on and may get reviewed, but must not be merged unless agreed explicitly.
- [x] Check the `needs documentation` label for any outstanding PRs flagged for this release and remove that label if done.
- [x] Check ./Jamulus -h output against the [Include-Client](https://github.com/jamulussoftware/jamuluswebsite/blob/next-release/wiki/en/Include-Client-Commands.md)/[Server-Commands](https://github.com/jamulussoftware/jamuluswebsite/blob/next-release/wiki/en/Include-Server-Commands.md).md pages and [man page (distributions/Jamulus.1)[(https://github.com/jamulussoftware/jamulus/blob/main/distributions/Jamulus.1). Update if necessary.
- [x] Start Website translations
  - [x] [Check for broken links](https://github.com/jamulussoftware/jamuluswebsite/blob/release/contribute/en/Release-Process.md#website-check-links)
  - [x] Open a Pull Request from `next-release` to release, set it as "Draft", sanity check for conflicts and any obvious problems.
  - [x] Declare a full freeze of the `next-release` and `release` branch. No changes should be made from now on to ensure translators don't have to work twice.
  - [ ] Check if the list of translators in `tools/create-translation-issues.sh`. Make sure issue text is up-to-date. Add any URLs that will need localisation into the "New/Changed screenshots" section.
  - [x] Create a translation issue for each language with `tools/create-translation-issues.sh` using `web` argument (see notes in script).
  - [x] If anyone finds critical issues now, all translators must be made aware of them and all languages should be updated.
- [x] Start App translations
  - [x] Generate `.ts` files in main via `lupdate`
  - [ ] Check if the list of translators in `tools/create-translation-issues.sh` is up-to-date
  - [x] Create a translation issue for each language with `tools/create-translation-issues.sh` using `app` argument.
- [x] [Update the Changelog](https://jamulus.io/contribute/Release-Process#updating-the-changelog)
- [x] [Tag a beta release](https://jamulus.io/contribute/Release-Process#steps-for-a-specific-release)
  - [x] Inform emlynmac for signing on macOS, and upload signed binary from [his repo](https://github.com/emlynmac/jamulus/releases/) to [ours](https://github.com/jamulussoftware/jamulus/releases/)
  - [x] Announce the beta release on Github Discussions. Pin the thread.
  - [x] Get feedback on the stability and resource usage (memleaks?, crashes?, installation issues?) of the beta release
- [x] Finish App translations
  - [x] Review translation PRs according to [release process checklist](https://jamulus.io/contribute/Release-Process#3-update-the-ts-files-returned-by-translators)
  - [x] Wait for all PRs to be merged (missing translations will revert to English automatically).
  - [ ] Check for conflicting accelerator keys (see `tools/checkkeys.pl`)
- [x] Finish Website translations
  - [x] Wait for all PRs to be merged (missing translations will revert to English automatically)
  - [ ] [Check for broken links](https://github.com/jamulussoftware/jamuluswebsite/blob/release/contribute/en/Release-Process.md#website-check-links)
- [x] Check the milestone for mergable stuff again
- [x] [Update the Changelog](https://jamulus.io/contribute/Release-Process#updating-the-changelog)
- [x] [Tag a release candidate](https://jamulus.io/contribute/Release-Process#steps-for-a-specific-release) (push the tag to [ann0see/jamulussign](https://github.com/ann0see/jamulussign/) for signing on macOS and upload signed binary to the official one).
  - [x] Announce the release candidate on Github Discussions. Pin the thread. Unpin and lock the beta thread.
  - [x] Draft an announcement, include all contributors via `tools/get_release_contributors.py`
- [x] [Update the Changelog](https://jamulus.io/contribute/Release-Process#updating-the-changelog)
- [x] [Tag the release version](https://jamulus.io/contribute/Release-Process#steps-for-a-specific-release):
  - [x] Update the version number in `Jamulus.pro` and add the release date to the Changelog header and commit.
  - [x] Wait for the release build to complete.
  - [x] Push the release to [ann0see/jamulussign](https://github.com/ann0see/jamulussign/) for macOS signing and upload signed binary [from the signing repo](https://github.com/ann0see/jamulussign/releases) repo [to the official one](https://github.com/jamulussoftware/jamulus/releases).
  - [x] Do a smoke test for Windows/Mac/Linux -- Do the binaries start/connect properly? Can earlier Jamulus versions properly connect to a server based on the new release?
  - [ ] Do a smoke test of the upgrade notification (do this _very quickly_)
    - [ ] Install the release build on _one_ of the update notification Servers
    - [ ] Run the _previous release_ (not an intermediate) build of GUI Client and Server
    - [ ] Check the update notice is displayed correctly
    - [ ] Revert the update notification Server to the previous release
  - [x] [Force tag that tag as `latest` and push.](https://jamulus.io/contribute/Release-Process#if-this-is-a-proper-release-move-the-latest-tag)
  - [ ] [Upload the artifacts to SourceForge and set defaults](https://jamulus.io/contribute/Release-Process#upload-binaries-to-sourceforge).
  - [x] Update download links on the website by editing `_config.yml` in `next-release`
  - [x] Disable branch protection rule of the `release` branch by clicking on "Edit" on the [Branches page](https://github.com/jamulussoftware/jamuluswebsite/settings/branches) and adding a `_` behind `release`.  
  - [x] Publish Website release by **squashing** and merging `next-release` into `release`
  - [x] Enable branch protection rule of the `release` branch after the site and the `.po` files are published by removing the `_` from the branch protection rule you edited on the [Branches page](https://github.com/jamulussoftware/jamuluswebsite/settings/branches).
- [x] Announce the new release with a summary of changes (+ link to the changelog for details) and a link to the download page
    - [x] On Github Discussions in the Announcements section. Lock the announcement thread. Pin the thread. Unpin and lock release candidate thread.
    - [x] On Facebook in the group "Jamulus (official group)". Turn off replies.
- [ ] Trigger the update notification by updating both Update Check Servers with the new version (@pljones for update02, email corrados for update01)
- [x] Prepare `Jamulus.pro` (`dev` suffix) and ChangeLog (add a header) for the [next release](https://jamulus.io/contribute/Release-Process#make-the-main-branch-ready-for-post-release-development)
- [x] Make sure there are no open issues or PRs on the [Project Tracking Board](https://github.com/orgs/jamulussoftware/projects/5/views/5) for the milestone
- [x] Create a milestone for the next minor release in jamulus and jamuluswebsite repos
- [x] Change the [Project Tracking Board](https://github.com/orgs/jamulussoftware/projects/5/views/5) to the next release (i.e. change the filter and save the change)
- [x] Close the release milestone in both jamulus and jamuluswebsite repos
- [ ] Update this template in https://jamulus.io/contribute/Release-Process with any improvements if needed.
- [x] Unpin and close this issue
- [ ] Determine if a release retrospective is needed, create on Discussions if required

**Comment by anprdev** [comment]
Hi all,

I love the JSON-RPC from Jamulus and hope it is not really changed in the upcoming update.

**Comment by ann0see** [comment]
@pljones Could you please verify you sent a mail about discord a few minutes ago?

**Comment by pljones** [comment]
Yes.  I've got to check a gew other places too see where else I did it, too.

**Comment by ann0see** [comment]
Ok. Verified. @gilgongo should be able to help you.

**Comment by ann0see** [comment]
I'd say the goal here is to have a stable 3.12.0 (@pljones already said something like that somewhere). Afterwards for 4.0 we start potential larger restructurings.

**Comment by ann0see** [comment]
I would like to declare a feature freeze this weekend: 05.07.25.

**Comment by anprdev** [comment]
Ho,

What you mean this. Jamulus down?

 
 

Sent: Monday, June 30, 2025 at 6:31 PM

**Comment by ann0see** [comment]
@anprdev No. This means that from this date no new features for the upcoming version should be worked on. Every new feature worked on would need to be in a later version.

**Comment by anprdev** [comment]
Thanks for answering.
Cheers Roland
 

Sent: Tuesday, July 01, 2025 at 7:27 AM

**Comment by anprdev** [comment]
Top good to do that.
Can we also do something about to tell the server to go private with RPC?
Maybe wrong place to ask.

**Comment by anprdev** [comment]
> Concerning signing on macOS, I will probably need to renew my certificate - but before doing that, I would like to have an unsigned beta.

If you like I can help with testing.

**Comment by ann0see** [comment]
@jamulussoftware/maindevelopers 

I am planning to push the release forward and soon start the translation process. 

The only important outstanding issue is about accessibility: https://github.com/jamulussoftware/jamulus/issues/490

IMO we need to build a modified version for this, so it's not strictly speaking a blocker but needs to be done soon.

**Comment by anprdev** [comment]
Nice step by step.

**Comment by ann0see** [comment]
@softins I don't think that the lupdate step is needed anymore...

**Comment by ann0see** [comment]
TODO: Add step for checklist to set note for translation process on weblate.

**Comment by pljones** [comment]
> [x] Check ./Jamulus -h output against the [Include-Client](https://github.com/jamulussoftware/jamuluswebsite/blob/next-release/wiki/en/Include-Client-Commands.md)/[Server-Commands](https://github.com/jamulussoftware/jamuluswebsite/blob/next-release/wiki/en/Include-Server-Commands.md).md pages and [man page (distributions/Jamulus.1)[(https://github.com/jamulussoftware/jamulus/blob/main/distributions/Jamulus.1). Update if necessary.

"https://github.com/jamulussoftware/jamuluswebsite/blob/next-release/wiki/en/Include-Server-Commands.md" -- this does not exist, unfortunately.  There's some more detailed, scattered, information on what the options mean in the "Running a Server" document but not just the list of options.  I've marked it done on the basis of my attempt to check that everything matched but ... well, I couldn't say for certain.

> - [ ] Start Website translations
>    - [ ] [Check for broken links](https://github.com/jamulussoftware/jamuluswebsite/blob/release/contribute/en/Release-Process.md#website-check-links)

I got as far as `bundle exec jekyll serve` but `linkcheck --no-warnings http://localhost:4000 > linkcheck.log` fails -- there doesn't appear to be a `--no-warnings` option.  I tried using the docker version and that didn't work, so I gave up at that point.

**Comment by pljones** [comment]
Another change request for the changelog-helper:
```
$ grep SKIP ./tools/changelog-helper.sh
        if [[ "${title_suggestion_in_pr}" == "SKIP" ]]; then
```
Could do with this issuing a message.    I got
```
Checking if all PR references in git log since r3_11_0 are included for 3.12.0 based on ref origin/main...
-> Ignoring PR #3512, which was mentioned in 'git log r3_11_0..HEAD', but already has milestone 'Release 3.11.0'
```
on the first pass.  I checked #3512 and it was mis-tagged, so I fixed it.  When I reran, I didn't see it mentioned.  Having
```
-> Skipping PR #3512 (...title...)
```
would have been more reassuring.

**Comment by ann0see** [comment]
@jamulussoftware/translators please re check your translations. I plan to merge translations this evening.

**Comment by ann0see** [comment]
Translations are now merged.  Website needs to be updated with screenshots. Maybe we tag a new beta tomorrow?

**Comment by pljones** [comment]
> - [ ] Finish App translations
>     - [ ] Review translation PRs according to [release process checklist](https://jamulus.io/contribute/Release-Process#3-update-the-ts-files-returned-by-translators)
>     - [ ] Wait for all PRs to be merged (missing translations will revert to English automatically).
>     - [ ] Check for conflicting accelerator keys (see tools/checkkeys.pl)
> - [ ] Finish Website translations
>     - [ ] Wait for all PRs to be merged (missing translations will revert to English automatically)
>     - [ ] [Check for broken links](https://github.com/jamulussoftware/jamuluswebsite/blob/release/contribute/en/Release-Process.md#website-check-links)

OK, so another thing we need to fix in the release process template is cutting another beta between `Finish App translations` and `Finish Website translations` so that the beta can be used for the screenshots.

**Comment by anprdev** [comment]
I think there is a small issue with setDirectory.

When I set the server to None on the server itself. So Directory ComboBox value is None.
I set the Directory with RPC to Any Genre 1 all is working perfect.
But when I reboot the Server PI5 the directory is back to None and Directory is not registered even I have set the Directory to Any Genre 1.

It is not a big deal if you know that.

**Comment by pljones** [comment]
How are you starting the server?  If you're _not_ using the GUI, the persisted settings will _not_ be loaded.

**Comment by anprdev** [comment]
I always start the Server with GUI. Always like to see what is happening on the screen.

**Comment by pljones** [comment]
OK, the problem, then, is likely that the RPC setting doesn't get to the UI and then when the system shuts down, the UI setting gets sent to the main program and then on to be stored.  So all part of the same issue with the RPC/<->/core/<->/UI interfaces.

**Comment by anprdev** [comment]
Again no issue for me when you know it.

**Comment by pljones** [comment]
https://github.com/jamulussoftware/jamulus/actions/runs/17714052632
The merge of the latest translations doesn't build all the versions.  Do we need them all to get a full set of screenshots -- is anything platform-specific that wouldn't be in one of the other builds?

**Comment by anprdev** [comment]
Nice release but missing aarch64 
jamulus_3.12.0_ubuntu_armhf.deb

**Comment by ann0see** [comment]
> https://github.com/jamulussoftware/jamulus/actions/runs/17714052632
> The merge of the latest translations doesn't build all the versions.  Do we need them all to get a full set of screenshots -- is anything platform-specific that wouldn't be in one of the other builds?

I think the draft PR should build it.

**Comment by ann0see** [comment]
@jamulussoftware/maindevelopers I hope that we can soon freeze the website.

**Comment by ann0see** [comment]
@jamulussoftware/translators @jamulussoftware/trreviewers @jamulussoftware/maindevelopers 

The translation phase of the website is now starting. We hope that no changes are to be made any more.

**Comment by pljones** [comment]
Have we got the accessible build done?  Or should we do an RC without that?  I think if it's the release candidate it should be complete, so I'd prefer to include the accessible build if it'll be in the final release for 3.12.0.

**Comment by pljones** [comment]
OK, ran the link checker (I don't like doing `--network host`... no _simple_ alternative, though):
```
$ docker run --network host --rm tennox/linkcheck http://192.168.1.19:8022/ > linkcheck.log
```
Output looks okay:
```
$ cat linkcheck.log
Crawling...

Stats:
    3278 links
      71 destination URLs
     774 URLs ignored
       0 warnings
       0 errors
```

**Comment by softins** [comment]
There is also an unresolved issue regarding the version update of ASIO SDK, where the auto-generated PR doesn't build because Steinberg has changed the zip file naming convention and contents directory. I've done some work on that, but still need to finish it off.

**Comment by pljones** [comment]
> There is also an unresolved issue regarding the version update of ASIO SDK, where the auto-generated PR doesn't build because Steinberg has changed the zip file naming convention and contents directory. I've done some work on that, but still need to finish it off.

Have we got an issue / PR for that?  I remember something but I must have forgotten to put it on the project board.

**Comment by ann0see** [comment]
@pljones No, I believe @chigkim had a working patch but it wasn't possible to integrate it in the main version as it broke the GUI somehow.

**Comment by ann0see** [comment]
https://github.com/jamulussoftware/jamulus/commit/d05319fc455dd5cecb0e8b69e8515d9770f73644

**Comment by pljones** [comment]
OK, I've dropped it from 3.12.0: https://github.com/jamulussoftware/jamulus/issues/490#issuecomment-3508073263

**Comment by pljones** [comment]
OK, I rebased the Qt update:
```
$ git diff
diff --cc .github/workflows/autobuild.yml
index b4e61199,1f31055f..00000000
--- a/.github/workflows/autobuild.yml
+++ b/.github/workflows/autobuild.yml
@@@ -248,8 -250,8 +248,13 @@@ jobs

            - config_name:            iOS (artifacts)
              target_os:              ios
++<<<<<<< HEAD
 +            building_on_os:         macos-15
 +            base_command:           QT_VERSION=6.7.3 ./.github/autobuild/ios.sh
++=======
+             building_on_os:         macos-14
+             base_command:           QT_VERSION=6.8.1 ./.github/autobuild/ios.sh
++>>>>>>> 17696309 (Build: Bump Qt6 from 6.7.3 to 6.8.1)
              # Build failed with CodeQL enabled when last tested 03/2022 (#2490).
              # There are no hints that iOS is supposed to be supported by CodeQL.
              # Therefore, disable it:
```
Resolving that left nothing else to update, so the PR closed itself.

**Comment by pljones** [comment]
Beta4 arriving here: https://github.com/jamulussoftware/jamulus/releases/tag/r3_12_0beta4

(Ugh... I didn't do a great job on the ChangeLog...)

**Comment by pljones** [comment]
@ann0see we can't cut the RC until
* we've no open PRs for 3.12.0 on jamulussoftware/jamulus
* we've frozen the translations for the app, pushed from Weblate and merged here

**Comment by pljones** [comment]
>  Update this template in https://jamulus.io/contribute/Release-Process with any improvements if needed.

That's not a helpful link - it should link to the repo/file, _that's_ what will have improvements made to it.

**Comment by pljones** [comment]
So here's what I got Claude to write.  It picked up on the change to `--ctrlmidich` from my comment in the PR.  I'm wondering if we could start including "Update the Release Annoucement" as part of the PR merge process (manually feed the latest tracking Release Annoucement and the PR into Github CoPilot and get it to suggest the updated).

---

---

# Jamulus 3.12.0 — What's New

> **Release candidate.** This is a release candidate for 3.12.0. The current stable release remains 3.11.0.

---

## For everyone

The Connect dialog now shows the version of each server in the list, so you can tell at a glance whether a server is running current software. You can also sort your mixer channels alphabetically — handy when you're in a busy session and trying to find a specific person quickly.

Pan controls now respond more smoothly. A subtle timing issue that caused lag when adjusting stereo positioning has been fixed, so what you move is what you hear, immediately.

Jamulus has also quietly solved a long-running annoyance: your own channel in the mixer is now always channel 0, rather than being assigned whatever slot the server happened to give you. This makes your own fader reliably stable across reconnects.

A link to Jamulus's privacy policy has been added to the application.

---

## For Windows users

Windows users with ASIO audio interfaces can now use MIDI controllers to manage Jamulus directly — no JACK required. This has been a gap for a while, and it opens up hardware control for the majority of Windows users who don't run a JACK setup.

Jamulus will also now prevent your PC from going to sleep or activating the screensaver while you're connected to a session. No more getting kicked mid-rehearsal because your computer decided to take a nap.

---

## MIDI gets a proper settings interface

MIDI control in Jamulus has always required editing command-line parameters — workable, but not exactly welcoming. This release adds a dedicated **MIDI tab** to the Settings window, where you can configure your MIDI input device, set controller assignments, and enable or disable MIDI input, all without touching the command line.

Better still, there's now a **MIDI Learn** function: click the Learn button next to any control, move a knob or fader on your hardware, and Jamulus stores the mapping automatically. No more looking up CC numbers.

A **pick-up mode** is also included, which prevents the abrupt jumps you get when a physical controller is in a different position than the current software value — it waits until the hardware "catches up" before taking over.

### ⚠️ Breaking change for existing `--ctrlmidich` users

The default starting MIDI CC number has changed from **70 to 0**. If you use `--ctrlmidich` without explicit CC assignments, your controller mappings will shift. Review your configuration and update your CC numbers accordingly.

Existing `--ctrlmidich` command-line users are otherwise unaffected; those settings continue to work as before and take precedence over saved GUI settings when specified.

---

## For mobile users (iOS & Android)

The mobile interface has received significant attention in this release. iOS has been upgraded to Qt 6, bringing it in line with other platforms, and the **compact view is now the default** on both iOS and Android — a much better fit for smaller screens. iOS also gets a proper app icon, plus fixes for a connect dialog display issue, a crash when closing the chat window, and a hang that occurred when changing the app language.

---

## For macOS users

The minimum supported macOS version for the legacy build is now **High Sierra (10.13)**. If you're on Sierra (10.12), this will be the last release series to consider before upgrading your OS.

---

## For server operators

Two reliability improvements for Linux servers: Jamulus now waits for the network interface to be available before starting, which prevents failures on systems where the service was launching too early at boot. Swap memory usage is also now disabled, which is good practice for real-time audio — it prevents the OS from paging out memory mid-session.

### Deprecation notice

The `--htmlstatus` / `-m` server option has been deprecated and will be removed in a future release. If you're using this to monitor connected clients, the replacement is the JSON-RPC `jamulusserver/getClients` method. Now is a good time to plan that migration.

---

## Translations

Japanese (`ja_JP`) has been added as a supported language, and existing translations have received updates across multiple languages via Weblate.

---

*A big thanks to all contributors who made this release possible.*

**Comment by anprdev** [comment]
testing 3.12 Beta4 and I see that sometimes Jamulus Server crash and is Frozen.
Is happening 2 times until now.
Disconnect all clients from the server.
Also see sometimes that a client connect with no name just empty but is not shown in the Jamulus Client.
Maybe somebody has any idea.

**Comment by ann0see** [comment]
Can you please provide logs on the crash?

**Comment by softins** [comment]
> testing 3.12 Beta4 and I see that sometimes Jamulus Server crash and is Frozen.

@anprdev please could you mention:
- the platform you are running the server on? (Windows, Mac, Linux distro) And the OS version.
- whether you are running the server in GUI mode or headless?
- how long it takes the crash to happen after you started the server?
- any log messages you see?

Could you try the [beta5 release](https://github.com/jamulussoftware/jamulus/releases/tag/r3_12_0beta5) and let us know if it is any different?

**Comment by anprdev** [comment]
> Can you please provide logs on the crash?

Sorry no log yet.
Have enable it with beta5

**Comment by anprdev** [comment]
> > testing 3.12 Beta4 and I see that sometimes Jamulus Server crash and is Frozen.
> 
> [@anprdev](https://github.com/anprdev) please could you mention:
> 
>     * the platform you are running the server on? (Windows, Mac, Linux distro) And the OS version.
> 
>     * whether you are running the server in GUI mode or headless?
> 
>     * how long it takes the crash to happen after you started the server?
> 
>     * any log messages you see?
> 
> 
> Could you try the [beta5 release](https://github.com/jamulussoftware/jamulus/releases/tag/r3_12_0beta5) and let us know if it is any different?

The Server is running on Raspberry PI 5 8GB 500GB SSD Linux Server is Blues/Rock Genre Any 1
GUI mode
The time between the crash can be days or weeks
Log is now enable with beta5

This is my Server startup script
jamulus -s -T -F -u 20 --jsonrpcport 25112 --discononquit --htmlstatus /home/pi5/www/htmlstat.log --welcomemessage /home/pi5/JamulusStartup/Jamulus-Welcome-Short.html --jsonrpcsecretfile /home/pi5/JamulusStartup/RPC/secret.txt --log /home/pi5/JamulusStartup/JamServerLOG/jamserver.log

**Comment by pljones** [comment]
You'll need to capture the stdout and stderr - I'm pretty sure there's failure cases that wouldn't get logged.  Append `2>&1 | tee .../jamserver-console.log` or similar, to capture that.

**Comment by pljones** [comment]
@ann0see 
> You could comment on the closed issues. 

Sorry, just spotted that comment.  Did you mean these?
> feat: automated Release Announcement draft maintained by GitHub Copilot #3640

Just trying out what CoPilot can do and started out setting it up completely wrong.  I've now got it under a little more control (so it's only running on my repo).

----

Not blocking for the RC but blocking release, we still need the website docs:
https://github.com/orgs/jamulussoftware/projects/5/views/7
* #3600
* #3571 
* #3398 
* #3416 
* #3502

If any of those _don't_ require documentation, could the assignee please confirm / remove the label.
If PRs have been raised, that's good enough to get the label removed because it means the required docs are on the project board - but make sure the correct Project and Milestone are assigned to the PR.

**Comment by anprdev** [comment]
> You'll need to capture the stdout and stderr - I'm pretty sure there's failure cases that wouldn't get logged. Append `2>&1 | tee .../jamserver-console.log` or similar, to capture that.

Have made it this way. Hope this will be correct
```bash
#!/bin/bash 


echo "Start Jamulus Server!"

LOGDIR="/home/pi5/JamulusStartup/JamServerLOG"
mkdir -p "$LOGDIR"

LOGFILE="$LOGDIR/jamulus_$(date +%Y-%m-%d).log"

echo "Logging to: $LOGFILE"

jamulus -s -T -F -u 20 \
  --jsonrpcport 25112 \
  --discononquit \
  --htmlstatus "$LOGDIR/htmlstat.log" \
  --welcomemessage /home/pi5/JamulusStartup/Jamulus-Welcome-Short.html \
  >> "$LOGFILE" 2>&1
```

**Comment by anprdev** [comment]
> Can you please provide logs on the crash?
Using Beta5.
Yesterday Crash again.
I get the idea that somebody try to connect to the server and is not a Jamulus Client,

[jamulus_09-04-2026.log](https://github.com/user-attachments/files/26624272/jamulus_09-04-2026.log)

**Comment by ann0see** [comment]
Please also provide an excerpt of the syslog (you can use journalctl to view it) from around the time you experienced the crash.

**Comment by anprdev** [comment]
> Please also provide an excerpt of the syslog (you can use journalctl to view it) from around the time you experienced the crash.

Nothing there. After the crash I had to restart the Server.
Or I don't know how to use it. 

Nothing to find used
journalctl  --since "2026-04-09 00:00:10"
and
journalctl  --since "2026-04-09 00:00:10" | grep error

**Comment by ann0see** [comment]
Ok. So did the server hang? Maybe it's a memory leak?

**Comment by anprdev** [comment]
> Ok. So did the server hang? Maybe it's a memory leak?

The Server shown the clients on the server itself but they where disconnected.
The Clients on the Server are with No GUI.
Only 1 Client has a GUI.
The Server is not responding and pressing X to Shut down. When the Dialog appears for Shutdown or Cancel it will not work.
Server is Frozen and not responding.
Checking the memory on the Server there is no problem at all.
Raspberry Pi still working correct and there is no Memory issue.

Only option to get Jamulus running again is to restart the Raspberry

**Comment by ann0see** [comment]
This sounds like Jamulus Server hangs. This could be a deadlock. Could you please provide an exact description in another Issue, ping pljones, softins and me there. Moreover, it would be good if - once the server hangs - you can paste a screenshot using htop we may then see if any resource is saturated.

**Comment by anprdev** [comment]
> This sounds like Jamulus Server hangs. This could be a deadlock. Could you please provide an exact description in another Issue, ping pljones, softins and me there. Moreover, it would be good if - once the server hangs - you can paste a screenshot using htop we may then see if any resource is saturated.

Will do but can  take days or even weeks.
Server is running 24/7 but I reboot the Raspberry at least every week.
So I will not do that in this case.
Htop is also running.

**Comment by pljones** [comment]
```
2026-04-09 23:42:32, XX.YY.ZZ.AA, connected (6)
Server Registration Status update: Registration requested
Server Registration Status update: Registered
Server Registration Status update: Registration requested
Server Registration Status update: Registered
```
That's two registration requests since `2026-04-09 23:42:32, XX.YY.ZZ.AA, connected (6)`.  I don't remember if that's a UTC timestamp or local time.

Did you check whether the Jamulus server process is still running?  (I've had X apps exit with a fault leaving the UI displayed and X not cleaning up.)

If it is still running, does it have the network port(s) still open?

Oh - are you running traditional X or Wayland?

I'm assuming from the description and log you're running the server in UI mode rather than headless.

Have you tried running it headless?  Does it still crash?

**Comment by anprdev** [comment]
> ```
> 2026-04-09 23:42:32, XX.YY.ZZ.AA, connected (6)
> Server Registration Status update: Registration requested
> Server Registration Status update: Registered
> Server Registration Status update: Registration requested
> Server Registration Status update: Registered
> ```
> 
> That's two registration requests since `2026-04-09 23:42:32, XX.YY.ZZ.AA, connected (6)`. I don't remember if that's a UTC timestamp or local time.
> 
> Did you check whether the Jamulus server process is still running? (I've had X apps exit with a fault leaving the UI displayed and X not cleaning up.)
> 
> If it is still running, does it have the network port(s) still open?
> 
> Oh - are you running traditional X or Wayland?
> 
> I'm assuming from the description and log you're running the server in UI mode rather than headless.
> 
> Have you tried running it headless? Does it still crash?

The GUI is still there so I assume that the process is running and nobody can connect.
About the Port I don't know. 
Running X11
I have not run the Server headless. I always like to see what is happening.
I can try headless but then nothing too see.
The Pi is running headless and control by VLC
@ann0see @pljones

**Comment by ann0see** [comment]
netstat -ulpn should show the open udp ports (I don't know the command from the top of my head but it's likely it)

**Comment by anprdev** [comment]
> netstat -ulpn should show the open udp ports (I don't know the command from the top of my head but it's likely it)

Will check it when I have a crash
@ann0see @pljones

**Comment by anprdev** [comment]
Crash again 
ScreenShot HTop PID 5005 and some more

<img width="1779" height="946" alt="Image" src="https://github.com/user-attachments/assets/3f6de668-bb15-42ea-b4fa-a62096338301" />

netstat -ulpn

<img width="1058" height="415" alt="Image" src="https://github.com/user-attachments/assets/d99af541-b399-4f15-ba6a-f94dd6122457" />

Lets see if you can find something


Screen GUI
Client is empty 

<img width="1384" height="897" alt="Image" src="https://github.com/user-attachments/assets/39a25353-8c8b-40b7-806d-f5338dc7eaa2" />

<img width="605" height="690" alt="Image" src="https://github.com/user-attachments/assets/80503d73-9aca-4ab4-8cc6-1f44e32055a9" />

@pljones @ann0see

**Comment by pljones** [comment]
Try running it without the RPC front-end.

**Comment by anprdev** [comment]
Try running it without the RPC front-end.

This is not running on a RPC Front-end or I don't understand what you mean about this.
Or you mean JSON-RPC?
The JSON-RPC is used only to set the Server Private or Public again that's all.
This is not uses very often.
I will disable this and see what is happening.

Also this morning Crash again.

@pljones @ann0see

**Comment by pljones** [comment]
The Jamulus Multi-track Player window is not part of Jamulus and, as far as I know, uses the experimental RPC interface.

I don't know anything about how it works but removing it means we can eliminate something unknown.

**Comment by anprdev** [comment]
> The Jamulus Multi-track Player window is not part of Jamulus and, as far as I know, uses the experimental RPC interface.
> 
> I don't know anything about how it works but removing it means we can eliminate something unknown.

I have disable it for the Jamulus Server. So I can not set it to private per JSON-RPC anymore. Let see what is happening.
The Jamulus Client is working perfect with it getting the userinfo needed who is logged in or out.


@pljones @ann0see

**Comment by anprdev** [comment]
Maybe I have to explain some more about the Multi-track player.
The Player comes with a Website where users can select Songs and with 2x click on the song they put it in a Wish-list to hear it next and can Listen or Jam with the Song. Song is removed from the Wish-list when it is playing.

The Software on the Raspberry PI 5 has two applications written with B4J (Java).
The Multi-track Player and the Web-socked Web-server.
The Player is control Jamulus so I can send messages to the Jamulus Chat like welcome the user and send sheet URL's for the playing song.
Also the Player send messages to the Web-server so this can shown on the Web-side.
The Web-server is talking to the Player who is control the Audio.

The Web-site can also be used as Jam Player this means that the user has the option, after login by Request, to control the Player itself.  Lets say Mute Bass-track will mute the Bass for all Jammers.
Jam-list can be saved for using it again.
Wish-list is changing to Jam-list no songs are removed from there if filled. Playing song go to the end of the list to Jam or listen to it again. Only 1 user has control over this function even if there are about 5 users. Band-playing I call it.
In this mode the Control-user also has the option if there are about minimum 3 Jammers to set the Jamulus Server to Private (what is disabled at the moment).  This Request is also send to the Player who will handle it.
If the Control-user leaves Jamulus, Server is set back to Public and Jam-List is Wish-list again.
Thats the only function I use for the Jamulus Server.

Hope that this can explain some more.
Any question please ask.
@pljones @ann0see

**Comment by pljones** [comment]
Something weird just happened.  My client issued the Update Available notice - neither of the update servers have had a new version though...

**Comment by anprdev** [comment]
> Something weird just happened. My client issued the Update Available notice - neither of the update servers have had a new version though...

Funny then.

**Comment by anprdev** [comment]
> Something weird just happened. My client issued the Update Available notice - neither of the update servers have had a new version though...

Get the same here in red. lol

<img width="706" height="553" alt="Image" src="https://github.com/user-attachments/assets/352f7a9e-09aa-4d4b-9954-12386517796c" />

**Comment by ann0see** [comment]
We've enabled IPv6 on the main directory. Maybe that explains it?

**Comment by ann0see** [comment]
Or @pljones does your second updatecheck have a newer version?

**Comment by pljones** [comment]
We've two DNS entries:
* updatecheck1.jamulus.io
* updatecheck2.jamulus.io

You can see in explorer.jamaulus.io that they're both 3.11.0.
<img width="514" height="48" alt="Image" src="https://github.com/user-attachments/assets/b2f77743-5c93-4f15-8372-160fcda4c122" />

<img width="510" height="43" alt="Image" src="https://github.com/user-attachments/assets/0c4e8ef9-8c09-41d8-91e3-2201c2313376" />

**Comment by ann0see** [comment]
And the local version is which one exactly for both of you?

**Comment by anprdev** [comment]
> And the local version is which one exactly for both of you?

3.12.0 Beta5 for Server and Client

**Comment by ann0see** [comment]
Weird.

**Comment by pljones** [comment]
Client
<img width="229" height="26" alt="Image" src="https://github.com/user-attachments/assets/c32ffb22-8efc-4ab2-9d8a-ed40f13ffb8c" />
Server
```
 *** Jamulus, Version 3.11.0dev-1b3acba5
```

**Comment by ann0see** [comment]
Maybe the last change with the time of the build broke the version check somehow.

**Comment by pljones** [comment]
Hm.  Don't think so -- I've been running this version since some time in March (before
```
 8d1cf470 2026-03-13 Merge pull request #3502 from ignotus666/midi-gui-learn [ann0see]
```
at least -- it from ignotus666's builds, I think) and it's not done it before.  I know the servers in question haven't been touched (based on their reported versions).

I'm wondering now what happens if the version check fails - do we fall back to saying "try getting a new version" (on the basis the new version broke the old version check)?  In which case, it could just have been a corrupt packet that got through the IP layer checks.

**Comment by ann0see** [comment]
As far as I remember you don't have IPv6 @pljones ?

**Comment by pljones** [comment]
No.  Not available through our ISP.

**Comment by ann0see** [comment]
I believe you could sign up for a trial: https://www.ispreview.co.uk/index.php/2025/04/broadband-isp-plusnet-uk-set-to-trial-ipv6-on-high-touch-network.html

(Or wait until they have spots available)

**Comment by pljones** [comment]
Nope, not here we can't.  That "only a small portion of their base are currently using the provider’s High Touch network" means "we've not actually rolled anything out except to some of our mates without the management knowing", I think.  (It did last time they tried it.)

**Comment by ann0see** [comment]
I'd like to tag a RC.

**Comment by pljones** [comment]
I've no objection...

----

The version update thing is weird.
```c++
#if ( QT_VERSION >= QT_VERSION_CHECK( 5, 6, 0 ) ) && !defined( DISABLE_VERSION_CHECK )
    int            mySuffixIndex;
    QVersionNumber myVersion = QVersionNumber::fromString ( VERSION, &mySuffixIndex );

    int            serverSuffixIndex;
    QVersionNumber serverVersion = QVersionNumber::fromString ( strVersion, &serverSuffixIndex );

    // only compare if the server version has no suffix (such as dev or beta)
    if ( strVersion.size() == serverSuffixIndex && QVersionNumber::compare ( serverVersion, myVersion ) > 0 )
    {
        // show the label and hide it after one minute again
        lblUpdateCheck->show();
        QTimer::singleShot ( 60000, [this]() { lblUpdateCheck->hide(); } );
    }
#endif
```
* `strVersion.size() == serverSuffixIndex` _disables_ checking if the update server has had a non-release version deployed to it.
* `QVersionNumber::compare` compares two QVersionNumber objects v1 and v2 and returns an integer less than, equal to, or greater than zero if v1 is less than, equal to, or greater than v2, respectively.
  * Because `QVersionNumber::fromString` was called on both "server" and "my" values, it _should_ work.  Unless it's broken in Qt 6.10.2.

The Connect Dialog does actually ping every server for the version... but regardless, if the server isn't running a release version, its response is ignored...
```c++
    // first request the server version if we have not already received it
    if ( bNeedVersion )
    {
        emit CreateCLServerListReqVerAndOSMes ( haServerAddress );
    }
```

**Comment by ann0see** [comment]
Tagged RC1 now.

**Comment by ann0see** [comment]
> include all contributors via tools/get_release_contributors.py

I probably got hit by an API limit...

**Comment by pljones** [comment]
Jazz, Rock, Classical/Folk and Choral/Barbershop Directories updated 3.12.0rc1; my private Directories and my jamulus.drealm.info Server to main and my private server to its rebased branch.

**Comment by anprdev** [comment]
> Tagged RC1 now.

Wondering where to download RC1.
I like to test it with my Server.
Running at the moment Beta 5

**Comment by softins** [comment]
> Wondering where to download RC1.

https://github.com/jamulussoftware/jamulus/releases

**Comment by anprdev** [comment]
> > Wondering where to download RC1.
> 
> https://github.com/jamulussoftware/jamulus/releases

Thanks
Have a nice weekend
@softins

**Comment by ann0see** [comment]
Linux build fails. This may just be a temporary hiccup.

**Comment by softins** [comment]
> Linux build fails. This may just be a temporary hiccup.

I've had a few random CI failures today.

**Comment by anprdev** [comment]
> Linux build fails. This may just be a temporary hiccup.

On my Raspberry PI 5 (Linux) I don't have any issue for now.

**Comment by ann0see** [comment]
So CI is fine now - I believe.

**Comment by ann0see** [comment]
Concerning the website, we may need to fix any broken links afterwards. Locally it shows nothing...

**Comment by ann0see** [comment]
I plan to do the final release very soon - tagging latest and pushing whould happen a bit later.

**Comment by ann0see** [comment]
Release is building :https://github.com/jamulussoftware/jamulus/releases/tag/r3_12_0

**Comment by anprdev** [comment]
> Release is building :https://github.com/jamulussoftware/jamulus/releases/tag/r3_12_0

Missing jamulus_3.12.0_ubuntu_armhf.deb @ann0see

**Comment by ann0see** [comment]
Yes. It's still building. Also there's an error for launchpad.net (Ubuntu PPA) which makes the linux builds fail - as soon as all other builds are done, I'll kick the linux builds off again.

macOS works (confirmed), Linux self build also works.

**Comment by anprdev** [comment]
> Yes. It's still building. Also there's an error for launchpad.net (Ubuntu PPA) which makes the linux builds fail - as soon as all other builds are done, I'll kick the linux builds off again.
> 
> macOS works (confirmed), Linux self build also works.

So I was too fast. lol
Thanks

**Comment by ann0see** [comment]
You could test Windows. I'm currently stuck with Linux and macOS.

**Comment by anprdev** [comment]
> You could test Windows. I'm currently stuck with Linux and macOS.

No Problem

Can do also on Windows.
@ann0see

**Comment by ann0see** [comment]
iOS will take some time since it needs to go through the AppStore review process - I hope that they don't find complaints.

**Comment by ann0see** [comment]
It's likely that we'll have to wait (maybe even a few days) until the ubuntu build succeeds...

**Comment by anprdev** [comment]
> It's likely that we'll have to wait (maybe even a few days) until the ubuntu build succeeds...

No problem at all.
Just take your time for it.
@ann0see

**Comment by ann0see** [comment]
It seems launchpad was under a cyber attack: https://status.canonical.com/#/incident/KNms6QK9ewuzz-7xUsPsNylV20jEt5kyKsd8A-3ptQEHpOd8VQ40ZQs-KD81fboQXeGZB94okNHdHBGlCv58Sw==

So probably best to wait it out.

**Comment by ann0see** [comment]
Uploaded the amd64 builds with a local build and reset the latest release to 3.11.0. I hope we'll be able to build the binaries again later then. The issue is that if 3.12.0 has not built on the CI, the debian repo will likely not work. Anyway, 3.12.0 is current stable but official linux builds don't exist.

**Comment by anprdev** [comment]
> Uploaded the amd64 builds with a local build and reset the latest release to 3.11.0. I hope we'll be able to build the binaries again later then. The issue is that if 3.12.0 has not built on the CI, the debian repo will likely not work. Anyway, 3.12.0 is current stable but official linux builds don't exist.

Better later than with errors and not working.
i wait for it no problem.
you  all enjoy the weekend.

**Comment by ann0see** [comment]
iOS was approved.

**Comment by ann0see** [comment]
I've deleted the release and started a re-run: https://github.com/jamulussoftware/jamulus/releases/tag/r3_12_0

It succeeded a few minutes ago but wasn't a clean build.

**Comment by pljones** [comment]
What was wrong with it, out of interest?

**Comment by ann0see** [comment]
It didn't correctly delete the hand compiled binaries. I'll need to re-run it again I'm afraid as it failed again.

**Comment by ann0see** [comment]
I believe the current run will work.

**Comment by ann0see** [comment]
I believe it's done now.

**Comment by anprdev** [comment]
> I believe it's done now.

Good Job.   Thanks
armhf is working.
Will update the Server @ann0see

**Comment by ann0see** [comment]
Ok. Webpage also updated. I'll hand over to @pljones and @corrados to update the servers and finish the other open tasks.

**Comment by anprdev** [comment]
Hi Guys
With the 3.12 release is something wrong or maybe you know the issue.
When I start the Jamulus client on Windows 11 the client connected to the server but  is freezing and take a long time to recover this.
Sometimes I need to kill it.
Go back to jamulus_3.11.0dev-c699905_win. or the 3.12 RC version Problem is solved.
@ann0see @pljones

**Comment by softins** [comment]
> Hi Guys With the 3.12 release is something wrong or maybe you know the issue.

I've opened a new issue #3677 to discuss this. Please continue over there.
